ee/clickhouse/views/experiments.py

```diff
@@ -258,17 +258,19 @@ def update(self, instance: Experiment, validated_data: dict, *args: Any, **kwarg
         if extra_keys:
             raise ValidationError(f"Can't update keys: {', '.join(sorted(extra_keys))} on Experiment")
 
-        if "feature_flag_variants" in validated_data.get("parameters", {}):
-            if len(validated_data["parameters"]["feature_flag_variants"]) != len(feature_flag.variants):
-                raise ValidationError("Can't update feature_flag_variants on Experiment")
-
-            for variant in validated_data["parameters"]["feature_flag_variants"]:
-                if (
-                    len([ff_variant for ff_variant in feature_flag.variants if ff_variant["key"] == variant["key"]])
-                    != 1
-                ):
+        # if an experiment has launched, we cannot edit its variants anymore.
+        if not instance.is_draft:
+            if "feature_flag_variants" in validated_data.get("parameters", {}):
+                if len(validated_data["parameters"]["feature_flag_variants"]) != len(feature_flag.variants):
                     raise ValidationError("Can't update feature_flag_variants on Experiment")
 
+                for variant in validated_data["parameters"]["feature_flag_variants"]:
+                    if (
+                        len([ff_variant for ff_variant in feature_flag.variants if ff_variant["key"] == variant["key"]])
+                        != 1
+                    ):
+                        raise ValidationError("Can't update feature_flag_variants on Experiment")
```

Comment:
do we already have a test for this case?  Maybe worth adding if not?

Reply:
Good question, we already have a test for this scenario. 

https://github.com/PostHog/posthog/blob/f94599be20d415b8a1bf792d4e671bdef5217a33/ee/clickhouse/views/test/test_clickhouse_experiments.py#L528-L531